gno.land/cmd/gnoland/testdata/event.txtar

```diff
@@ -0,0 +1,42 @@
+# start a new node
+gnoland start
+
+gnokey maketx addpkg -pkgdir $WORK -pkgpath gno.land/r/demo/ee -gas-fee 1000000ugnot -gas-wanted 2000000 -broadcast -chainid=tendermint_test test1
+stdout OK!
+
+gnokey maketx call -pkgpath gno.land/r/demo/ee -func Hello -gas-fee 1000000ugnot -gas-wanted 2000000 -broadcast -chainid=tendermint_test test1
+stdout OK!
+stdout 'GAS WANTED: 2000000'
+stdout 'GAS USED: [0-9]+'
+stdout 'EVENTS: \[.*?\]'
+
+-- ee.gno --
+package ee
+
+import (
+	"std"
+)
+
+const (
+	EventAttrKeySender = "sender"
+	EventAttrKeyReceiver = "receiver"
+)
+
+func Hello() {
+	std.EmitEvent(
+		EventAttrKeySender,
+		[]string{
+			"key1", "value1",
+			"key2", "value2",
+			"key3", "value3",
+			"not-a-pair",		// ignore the key-value not in pairs
+		},
+	)
+
+	std.EmitEvent(
+		EventAttrKeyReceiver,
+		[]string{
+			"foo", "bar",
+		},
```

Comment:
Have you considered and determined the impact of switching to a more type-safe approach here?

```suggestion
	std.EmitEvent(
		EventAttrKeyReceiver{
			Foo: "bar",
		},
	)
```

Reply:
Hmm.. then we could receive an `interface` as an parameter and handle it using `reflect`. Something like this:

```go
func X_emitEvent(m *gno.Machine, event interface{}) {
	val := reflect.ValueOf(event)
	typ := val.Type()

	if typ.Kind() != reflect.Struct {
		m.Panic(typedString("invalid type"))
	}

	typeName := typ.Name()
	numFields := typ.NumField()

	eventAttrs := make([]sdk.EventAttribute, 0, numFields)

	for i := 0; i < numFields; i++ {
		field := typ.Field(i)
		fieldName := field.Name
		fieldValue := val.Field(i).Interface()

		eventAttrs = append(eventAttrs, sdk.EventAttribute{
			Key:   fieldName,
			Value: fieldValue.(string),
		})
	}
    // ...
}
```

Reply:
@jaekwon wdyt?

Reply:
We can begin with the current approach and later transition to v2, which will involve breaking changes from the current implementation. Let's await PR #1776, which will bring in a reusable reflection feature. Following this, we can adopt a more type-safe approach, allowing for new patterns such as utilizing private structs created by public constructors.